tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -20,6 +29,218 @@ def test_hyperopt_parameters(use_hyperband):
     study.optimize(lambda t: t.suggest_uniform("x", 10, 20), n_trials=50)
 
 
+def test_sample_relative() -> None:
+    sampler = TPESampler()
+    # Study and frozen-trial are not supposed to be accessed.
+    study = Mock(spec=[])
+    frozen_trial = Mock(spec=[])
+    assert sampler.sample_relative(study, frozen_trial, {}) == {}
+
+
+def test_infer_relative_search_space() -> None:
+    sampler = TPESampler()
+    # Study and frozen-trial are not supposed to be accessed.
+    study = Mock(spec=[])
+    frozen_trial = Mock(spec=[])
+    assert sampler.infer_relative_search_space(study, frozen_trial) == {}
+
+
+def test_sample_independent() -> None:
+    study = optuna.create_study()
+    dist = optuna.distributions.UniformDistribution(1.0, 100.0)
+    past_trials = [frozen_trial_factory(i, dist=dist) for i in range(1, 8)]
+
+    # Prepare a trial and a sample for later checks.
+    with patch("optuna.Study.get_trials", return_value=past_trials):
```

Comment:
Could you place operations that are not dependent on `optuna.Study.get_trials` outside this context manager? Above all, it aids mental readability.